dotnet/src/SemanticKernel.Core/Functions/InstrumentedSKFunction.cs

```diff
@@ -18,153 +18,92 @@
 namespace Microsoft.SemanticKernel;
 #pragma warning restore IDE0130
 
-/// <summary>
-/// Standard Semantic Kernel callable function with instrumentation.
-/// </summary>
+/// <summary>Instrumented function that surrounds the invocation of another function with logging and metrics.</summary>
 internal sealed class InstrumentedSKFunction : ISKFunction
 {
-    /// <inheritdoc/>
-    public string Name => this._function.Name;
+    /// <summary><see cref="ActivitySource"/> for function-related activities.</summary>
+    private static readonly ActivitySource s_activitySource = new(typeof(ISKFunction).FullName!);
```

Comment:
I feel like "Microsoft.SemanticKernel" would be a better name for the `ActivitySource` as well. To listen in OTel you would do something like...

```csharp
using var tracerProvider = Sdk.CreateTracerProviderBuilder()
   .AddSource("Microsoft.SemanticKernel")
   .AddMyFavoriteExporter()
   .Build();
```

As opposed to...

```csharp
using var tracerProvider = Sdk.CreateTracerProviderBuilder()
   .AddSource("Microsoft.SemanticKernel.ISKFunction")
   .AddMyFavoriteExporter()
   .Build();
```